web/ui/react-app/src/pages/Services.tsx

```diff
@@ -2,14 +2,113 @@ import React, { FC } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+interface DiscoveredLabels {
+  address: string;
+  metrics_path: string;
+  scheme: string;
+  job: string;
+  my: string;
+  your: string;
+}
+
+interface ActiveTargets {
+  discoveredLabels: DiscoveredLabels[];
+  labels: any;
+  scrapePool: string;
+  scrapeUrl: string;
+  lastError: string;
+  lastScrape: string;
+  lastScrapeDuration: number;
+  health: string;
+}
+
+interface ServiceMap {
+  activeTargets: ActiveTargets[];
+  droppedTargets: any[];
+}
+
+const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<ServiceMap>(`${pathPrefix}/api/v1/targets`);
+  const processTargets = (response: ServiceMap) => {
+    console.warn(response);
+    const activeTargets = response.activeTargets;
```

Comment:
You extract `activeTargets` into a separate const here, but not `droppedTargets`, although they are used in the same way below. Since you're accessing each only once, I'd probably omit this for `activeTargets`.